Support Java 11 (6.x)

cmark opened the pull request:
This PR introduces 7.x style target-platform project on 6.x and eliminates the need for a separate target platform build job to develop plug-ins/features for Snow Owl.
Additional changes:
* SLF4J API bumped to 1.7.25
* Logback bumped to 1.2.3
* Fastutil bumped to 8.2.2
* Tycho bumped to 1.2.0
* Spring bumped to 4.3.22
* Replaced swaggermvc with Springfox 2.8.0
* Removed all embedded dependencies (inside lib folders) and added as Maven dependency
* Update test launch configuration names

cmark committed:
[tp] share initial version of 7.x style target-platform project

cmark committed:
[core] backport snowowl.home configuration setting

Use snowowl.home in snow-owl-oss.launch config.

cmark committed:
[build] remove releng profiles

Remove system.bundle.package.exporter.
Remove target.update project.

cmark committed:
[build] remove kotlin version property

cmark committed:
[tp] bump logback and slf4j to 1.2.3 and 1.7.25

cmark committed:
[tp] move runtime folder to com.b2international.snowowl.server.update

Rename so_server.product to snow-owl-oss.product.
Copy users and so config to runtime folder.
Remove unnecessary launch configurations.

cmark committed:
[dev] remove unnecessary relative path from config copy operation

cmark committed:
[build] update server.dependencies.feature

cmark committed:
[dev] add Maven nature to validation.snomed

cmark committed:
[build] remove scg and semanticengine from build

cmark committed:
[build] fix build after target-platform changes

cmark committed:
[build] fix mvn build.launch

cmark committed:
[tests] update test launch configuration

Rename them to match 7.x launch config name convention.

Commit:
[dependency] Add javax.xml.bind, javax.activation dependencies to...

...resolve java 11 runtime errors

apeteri commented on core/com.b2international.snowowl.api.rest.tests/pom.xml:
```diff
@@ -23,22 +23,17 @@
 								<type>eclipse-plugin</type>
 								<id>org.eclipse.jetty.osgi.boot</id>
-								<versionRange>[8.1.0,8.2.0)</versionRange>
+								<versionRange>0.0.0</versionRange>
```
Is this needed because of the difference between the "server" and "server + client" target platform versions of Jetty?

cmark replied:
No, they use the same Jetty version now. 
This is purely to remove the extra headache when you update Jetty versions.

apeteri commented on core/com.b2international.snowowl.datastore.server.tests/src/com/b2international/snowowl/datastore/internal/branch/MockInternalCDOBranchManager.java:
```diff
@@ -48,7 +48,7 @@
 	public MockInternalCDOBranchManager(CDOTimeProvider clock) {
 		this.clock = clock;
-		when(delegate.getTimeProvider()).thenReturn(clock);
+//		when(delegate.getTimeProvider()).thenReturn(clock);
```
What is the reason for these commented lines?

cmark replied:
We can remove those, the new Mockito complains about them that they are unused stubs and should be removed.

apeteri replied:
👍

apeteri commented on net4j/com.b2international.snowowl.eventbus.tests/src/com/b2international/snowowl/internal/eventbus/net4j/EventBusProtocolTest.java:
```diff
@@ -144,7 +143,7 @@ public void testNotifyEvent_NullResult_ShouldBeHandledProperly() throws Exceptio
 	private void prepareEvent(boolean isAdded) {
 		when(event.getAddress()).thenReturn(ADDRESS);
-		when(event.getSource()).thenReturn(eb);
+//		when(event.getSource()).thenReturn(eb);
```
(same in this file)

cmark replied:
See comment above.

cmark committed:
[tp] remove rest api jars from git repository

cmark committed:
[tp] use Springfox 2.8.0 and Spring 4.3.22 in snowowl.api.rest

cmark committed:
[tp] update Swagger UI to Swagger 2.x compatible version

cmark committed:
[tp] update Api annotations

cmark committed:
[tp] use Springfox 2.8.0 and Spring 4.3.22 in snomed.api.rest

cmark committed:
[tp] update Swagger UI to Swagger 2.x compatible version

cmark committed:
[tp] update RestAssured to 3.2.0

cmark committed:
[tp] use fastutil library from target platform

cmark committed:
[tp] update mapdb to 3.0.7

cmark committed:
[tp] use bcrypt library from target platform

cmark committed:
[tp] use typetools from tp, remove flexigraph from commons

cmark committed:
[api] remove IAuthenticationService interface

cmark committed:
[tp] use maven repository to fetch OSGi bundles

Add JBcrypt.
Add Typetools.
Add HdrHistogram.
Add commons-logging.
Add commons-codec.
Add fastutil.
Add classmate.
Add mapstruct.
Add swagger-annotations.
Add swagger-models.
Remove unnecessary jackson dependencies.

cmark committed:
[launch] update launch configs

cmark committed:
Merge branch '6.x' into feature/backport-target-platform-project

Conflicts:
	releng/com.b2international.snowowl.server.target.update/pom.xml
	releng/com.b2international.snowowl.server.update/so_server.product
	releng/system.bundle.package.exporter/META-INF/MANIFEST.MF
	releng/system.bundle.package.exporter/pom.xml

cmark committed:
[build] update version of target-platform project...

...and snow-owl-oss.product.

cmark committed:
[build] upgrade Tycho to 1.2.0

cmark committed:
[protege] remove unused import statement

cmark committed:
[tp] update dependencies to be Java 11 compatible

cmark committed:
[tp] remove databinding dependency from commons

cmark committed:
[tp] bump protege to 5.0.1-b2i version

Bump owlapi to 4.5.10.

cmark committed:
[tp] update server.dependencies.feature

cmark committed:
[tp] remove javax.annotations package imports

Remove 0.4.1 version of ELK.
Remove jsr305 from commons.

cmark committed:
[tp] remove protege bundles

Use the 5.0.x-b2i version from Maven repository.

cmark committed:
[tp] bump protege to 5.0.2-b2i version

cmark committed:
[tp] remove jetty 8.1.14 from target platform

cmark committed:
[tp] remove javax.el, com.sun.el and classmate from hibernate wrapper

cmark committed:
[tp] bump Xtext to 2.17.0

Bump Guava to 21.0.
Bump EMF to 2.15.

cmark committed:
[tp] fix compile errors due to Guava 21.0 version bump

cmark committed:
[tp] start jetty.osgi.boot via test rule instead of launch config

cmark committed:
[tp] add effective time format to SnomedComponent...

...and to ISnomedBrowserComponent.

cmark committed:
[tp] remove jackson guava module from target platform

Update associationTargets of SnomedConcept and SnomedDescription to
still serialize into Multimap in JSON.
Update REST API update domain classes to accept Multimap like structure
instead of using Guava's Multimap.

cmark committed:
[tp] do NOT override default ontology factories in OwlOntologyManager

Use HasPriority annotation on DelegateOntologyFactory to prioritize
before the default ontology factories.

cmark committed:
[launch] update launch configurations

cmark committed:
[tp] update Guava version range

cmark commented:
Ping!
